src/prng/chacha.rs

```diff
@@ -62,45 +63,45 @@ const STATE_WORDS: usize = 16;
 ///      http://cr.yp.to/papers.html#xsalsa)
 ///
 /// [`set_rounds`]: #method.set_counter
-#[derive(Clone)]
-pub struct ChaChaRng {
-    buffer:  [u32; STATE_WORDS], // Internal buffer of output
-    state:   [u32; STATE_WORDS], // Initial state
-    index:   usize,              // Index into state
-    rounds:  usize,
-}
+#[derive(Clone, Debug)]
+pub struct ChaChaRng(BlockRng<ChaChaCore>);
```

Comment:
So the `RngCore` impl wraps the `BlockRng` one, the `SeedableRng` impl has two layers of wrappers, and `CryptoRng` is implemented directly (with an unused in-between wrapper). Not ideal. I wonder if it makes more sense to say `put type ChaChaRng = BlockRng<ChaChaCore>;` but that has its disadvantages. Or maybe we should try to support `#[derive(Rng)]`?

Reply:
I don't think it is ideal, but I can live with ~30 lines of boilerplate. If we re-export the type it is not possible to attach documentation, right? And I made the fields from `BlockRng` public expecting it to be wrapped...